test(longevities): duplicate tablets-supported tests for 2024.2

yarongilor opened the pull request:
The tablets feature is switched to be disabled by default in 2024.2.
    The relevant longevity tests that are supported by tablets should be
    duplicated with a tablets-enabled configuration.

Notes: 

- The `configuration_tablets_enabled.yaml` file name is not added to all pipelines under `loongevity_tablets` directory since not all of them are planned to be tested. (basically some files of this folder are not needed and can be removed)
- Not all SCT test directories/group are duplicated. 
- Each relevant directory is basically added in a separate commit.
- Not added directories are: artifacts, features, kafka, load, nemesis, raft, rolling-upgrade,scale, performance


### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [ ]

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

temichus commented:
@yarongilor @fruch do we really need such a big change to enable tablets? 
if it goes only in [scylladb:branch-2024.2](https://github.com/scylladb/scylla-cluster-tests/tree/branch-2024.2)

maybe we can just enable it by default in the config?

https://github.com/scylladb/scylla-cluster-tests/blob/f1e62309bc84eed29abfdc8af4d2f8fa6b6bef19/sdcm/provision/scylla_yaml/scylla_yaml.py#L345

yarongilor commented:
> @yarongilor @fruch @roydahan, do we really need such a big change to enable tablets? if it goes only in [scylladb:branch-2024.2](https://github.com/scylladb/scylla-cluster-tests/tree/branch-2024.2)
> 
> maybe we can just enable it by default in the config?
> 
> https://github.com/scylladb/scylla-cluster-tests/blob/f1e62309bc84eed29abfdc8af4d2f8fa6b6bef19/sdcm/provision/scylla_yaml/scylla_yaml.py#L345

@temichus , we can't enable it by default since we have to test both enabled and disabled tablets.
I now created a script that go over pipeline files in a directory and adds a tablets-enabled configuration to it:
The script will now be used to add some more required directories. If the script is correct, then no much worries about errors in this respect.
```
#!/usr/bin/env python3

import os
import re
import argparse

def update_jenkinsfile(directory_path):
    for root, _, files in os.walk(directory_path):
        for file in files:
            if file.endswith('.jenkinsfile'):
                file_path = os.path.join(root, file)
                
                with open(file_path, 'r') as f:
                    lines = f.readlines()
                
                updated_lines = []
                updated = False
                
                for line in lines:
                    match = re.match(r'^\s*test_config:\s*(\'\'\'\[.*\]\'\'\'|\'[^\']*\'|\"[^\"]*\")\s*,?\s*$', line)
                    if match:
                        config_value = match.group(1)
                        
                        if "configurations/tablets_enabled.yaml" not in config_value:
                            if config_value.startswith("'''["):
                                config_value = config_value[:-4] + ', "configurations/tablets_enabled.yaml"]\'\'\''
                            else:
                                config_value = f"'''[\"{config_value[1:-1]}\", \"configurations/tablets_enabled.yaml\"]'''"
                                
                            line = re.sub(r'^\s*test_config:\s*.*$', f'    test_config: {config_value}', line)
                            updated = True
                    
                    updated_lines.append(line)
                
                if updated:
                    with open(file_path, 'w') as f:
                        f.writelines(updated_lines)
                    print(f"Updated: {file_path}")
                else:
                    print(f"No update needed: {file_path}")

def main():
    parser = argparse.ArgumentParser(description="Update Jenkinsfile test_config with configurations/tablets_enabled.yaml")
    parser.add_argument('directory', type=str, help='The directory path to search for .jenkinsfile files')
    args = parser.parse_args()

    update_jenkinsfile(args.directory)

if __name__ == '__main__':
    main()
```

temichus reviewed (approved):
ok, let's start testing

yarongilor commented:
Added the following directories as well:

- 	enterprise/SCT_Enterprise_Features/EncryptionAtRest_tablets/
- 	enterprise/SCT_Enterprise_Features/FIPS_tablets/
- 	enterprise/SCT_Enterprise_Features/ICS_tablets/
- 	enterprise/SCT_Enterprise_Features/LDAP_tablets/
- 	enterprise/SCT_Enterprise_Features/Workload_Prioritization_tablets/
- 	enterprise/SCT_Enterprise_Features/audit_tablets/
- 	oss/raft_tablets/

fruch commented on jenkins-pipelines/oss/tier2_tablets/scale-5000-tables.jenkinsfile:
```diff
@@ -7,5 +7,6 @@ longevityPipeline(
     test_name: 'longevity_test.LongevityTest.test_user_batch_custom_time',
-    test_config: '''["test-cases/scale/longevity-5000-tables.yaml", "configurations/tablets_disabled.yaml"]'''
+    test_config: '''["test-cases/scale/longevity-5000-tables.yaml", "configurations/tablets_enabled.yaml"]'''
+    timeout: [time: 4440, unit: 'MINUTES'],
```
can you remove this extra timeout

fruch commented on jenkins-pipelines/oss/tier2_tablets/longevity-double-cluster-with-schema-changes.jenkinsfile:
```diff
@@ -0,0 +1,13 @@
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: '''["test-cases/longevity/longevity-change-cluster-size-by-2-times.yaml", "configurations/tablets_enabled.yaml"]''',
+
+    timeout: [time: 820, unit: 'MINUTES']
```
we can drop this timeout

fruch reviewed (approved):
LGTM

yarongilor commented:
Running the script of https://jenkins.scylladb.com/view/QA/job/QA-tools/job/create-sct-test-jobs-for-branch/427/ for updating the new jenkins jobs
